python/lsst/ts/externalscripts/base_parameter_march.py

```diff
@@ -0,0 +1,442 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["BaseParameterMarch"]
+
+import abc
+import asyncio
+import types
+
+import numpy as np
+import yaml
+from lsst.ts import salobj
+from lsst.ts.standardscripts.base_block_script import BaseBlockScript
+
+
+class BaseParameterMarch(BaseBlockScript):
+    """Perform a parameter march by incrementally adjusting a
+    specific combination of degrees of freedom. This process
+    involves moving the parameters in specified increments
+    to evaluate the system's response. The sensitivity matrix
+    is an example of this technique, where the system's behavior
+    is assessed by capturing data at various incremental adjustments
+    for each degree of freedom.
+
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+    **Checkpoints**
+
+    * Step 1/n_steps starting position: start_position.
+    * Step n/n_steps.
+    """
+
+    def __init__(self, index, descr="Perform a parameter march.") -> None:
+        super().__init__(index=index, descr=descr)
+
+        self.ocps = None
+
+        self.config = None
+        self.dofs = np.zeros(50)
+
+        self.total_offset = 0.0
+        self.iterations_started = False
+
+    @property
+    @abc.abstractmethod
+    def tcs(self):
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    async def configure_tcs(self):
+        """Abstract method to configure the TCS."""
+        raise NotImplementedError()
+
+    @property
+    @abc.abstractmethod
+    def camera(self):
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    async def configure_camera(self):
+        """Abstract method to configure the Camera."""
+        raise NotImplementedError()
+
+    async def configure_ocps(self):
+        """Configure the OCPS remote object."""
+        if self.ocps is None:
+            self.log.debug("Configuring remote for OCPS:101")
+            self.ocps = salobj.Remote(self.domain, "OCPS", 101)
+            await self.ocps.start_task
+        else:
+            self.log.debug("OCPS already configured. Ignoring.")
+
+    @classmethod
+    def get_schema(cls) -> dict:
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_externalscripts/base_parameter_march.yaml
+            title: BaseParameterMarch v1
+            description: Configuration for BaseParameterMarch.
+            type: object
+            properties:
+              az:
+                description: Azimuth position to point to.
+                type: number
+                minimum: 0
+                maximum: 360
+              el:
+                description: Elevation position to point to.
+                type: number
+                minimum: 0
+                maximum: 90
+              filter:
```

Comment:
I'm wondering on the merits of including `filter` as a parameter here, as opposed to simply always using the current filter (and relegating a filter change to a separate script as necessary).  Changing a filter is a fairly big deal for LSSTCam (though less so for ComCam), so it feels like it maybe shouldn't be buried in a separate script; it should be its own script.  I don't think the same applies to any of the other parameters here (the initial slew could also be a big deal, but since the script ratchets in between triplets I don't really see an alternative).  Curious what @tribeiro and others think.

Reply:
I am not sure if there has been a resolution for this, but FWIW I think the ability to change the filter like this is already present in other scripts. We can just leave it blank if we don't want to change it?

Reply:
You are right that changing the filter for LSSTCam is big deal. I have to think about this, for now, I would suggest we leave it here.